examples/ra-sim/sim.py

```diff
@@ -0,0 +1,209 @@
+import random
+import sys
+from typing import Callable, Dict, Tuple
+
+import gym
+import jax
+import jax.numpy as jnp
+from chex import dataclass, Array, Numeric, PRNGKey, Scalar
+
+from reinforced_lib.exts import IEEE_802_11_ax
+
+
+gym.envs.registration.register(
+    id='RASimEnv-v1',
+    entry_point='examples.ra-sim.sim:RASimEnv'
+)
+
+
+# CW constraints
+MIN_CW_EXP = 4
+MAX_CW_EXP = 10
+
+# mean value based on the ns-3 static simulation with 1 station, ideal channel, AMPDU on, and constant MCS
+FRAMES_PER_SECOND = 188
+
+# based on the ns-3 static simulation with 1 station, ideal channel, and constant MCS
+AMPDU_SIZES = jnp.array([3, 6, 9, 12, 18, 25, 28, 31, 37, 41, 41, 41])
+
+# default values used by the ns-3 simulator
+# https://www.nsnam.org/docs/models/html/wifi-testing.html#packet-error-rate-performance
+DEFAULT_NOISE = -93.97
+DEFAULT_TX_POWER = 16.0206
+
+# LogDistance channel model
+# https://www.nsnam.org/docs/models/html/propagation.html#logdistancepropagationlossmodel
+REFERENCE_SNR = DEFAULT_TX_POWER - DEFAULT_NOISE
+REFERENCE_LOSS = 46.6777
+EXPONENT = 3.0
+
+
+def distance_to_snr(distance: Numeric) -> Numeric:
+    return REFERENCE_SNR - (REFERENCE_LOSS + 10 * EXPONENT * jnp.log10(distance))
+
+
+@dataclass
+class RASimState:
+    time: Array
+    snr: Array
+    ptr: jnp.int32
+    cw: jnp.int32
+
+
+@dataclass
+class Env:
+    init: Callable
+    step: Callable
+
+
+def ra_sim(
+        simulation_time: Scalar,
+        velocity: Scalar,
+        initial_position: Scalar,
+        n_wifi: jnp.int32,
+        total_frames: jnp.int32
+) -> Env:
```

Comment:
Uważasz, że warto do tej funkcji i jej metod (`init` i `step`) dopisywać docstringi czy nie ma co? Nie przychodzi mi na myśl nic, co mógłbym napisać oprócz kopiuj-wklej tego, co napisałem poniżej w metodach klasy `RASimEnv`.

Reply:
Chyba nie ma co tam dopisywać na siłę, szczególnie że i tak je przykrywa `jax.jit`, więc nie będą się podświetlały w IDE.

Reply:
Racja, to zostawiam tak.